pymc3/distributions/continuous.py

```diff
@@ -4026,3 +4045,201 @@ def logcdf(value, mu, sigma):
             at.log(at.erfc(at.exp(-scaled / 2) * (2 ** -0.5))),
             0 < sigma,
         )
+
+
+class PolyaGammaRV(RandomVariable):
+    """Polya-Gamma random variable."""
+
+    name = "polyagamma"
+    ndim_supp = 0
+    ndims_params = [0, 0]
+    dtype = "floatX"
+    _print_name = ("PG", "\\operatorname{PG}")
+
+    def __call__(self, h=1.0, z=0.0, size=None, **kwargs):
+        return super().__call__(h, z, size=size, **kwargs)
+
+    @classmethod
+    def rng_fn(cls, rng, h, z, size=None):
+        """
+        Generate a random sample from the distribution with the given parameters
+
+        Parameters
+        ----------
+        rng : {None, int, array_like[ints], SeedSequence, BitGenerator, Generator}
+            A seed to initialize the random number generator. If None, then fresh,
+            unpredictable entropy will be pulled from the OS. If an ``int`` or
+            ``array_like[ints]`` is passed, then it will be passed to
+            `SeedSequence` to derive the initial `BitGenerator` state. One may also
+            pass in a `SeedSequence` instance.
+            Additionally, when passed a `BitGenerator`, it will be wrapped by
+            `Generator`. If passed a `Generator`, it will be returned unaltered.
+        h : scalar or sequence
+            The shape parameter of the distribution.
+        z : scalar or sequence
+            The exponential tilting parameter.
+        size : int or tuple of ints, optional
+            The number of elements to draw from the distribution. If size is
+            ``None`` (default) then a single value is returned. If a tuple of
+            integers is passed, the returned array will have the same shape.
+            If the element(s) of size is not an integer type, it will be truncated
+            to the largest integer smaller than its value (e.g (2.1, 1) -> (2, 1)).
+            This parameter only applies if `h` and `z` are scalars.
+        """
+        # handle the kind of rng passed to the sampler
+        bg = rng._bit_generator if isinstance(rng, np.random.RandomState) else rng
+        return random_polyagamma(h, z, size=size, random_state=bg)
+
+
+polyagamma = PolyaGammaRV()
+
+
+class _PolyaGammaLogDistFunc(Op):
+    __props__ = ("get_pdf",)
+
+    def __init__(self, get_pdf=False):
+        self.get_pdf = get_pdf
+
+    def make_node(self, x, h, z):
+        x = at.as_tensor_variable(floatX(x))
+        h = at.as_tensor_variable(floatX(h))
+        z = at.as_tensor_variable(floatX(z))
+        shape = broadcast_shape(x, h, z)
+        broadcastable = [] if not shape else [False] * len(shape)
+        return Apply(self, [x, h, z], [at.TensorType(np.float64, broadcastable)()])
```

Comment:
Does the output have to be `float64`? Again users might want to run everything in `float32`. If for some reason that can't be supported, we should just raise an informative error.

Reply:
I missed that line. Thanks for pointing it out. resolved in latest commit.